src/newuidmap.c

```diff
@@ -158,9 +159,9 @@ int main(int argc, char **argv)
 	 * mappings we have been asked to set.
 	 */
 	if ((getuid() != pw->pw_uid) ||
-	    (getgid() != pw->pw_gid) ||
+	    (!getdef_bool("GRANT_AUX_GROUP_SUBIDS") && (getgid() != pw->pw_gid)) ||
```

Comment:
Sorry this should have occurred to me before.

You only need to support the user doing newgrp and then starting a container, right?

So can we (in both of these checks) still enforce that regid == egid? i.e.
    getresgid(&r, &e, &s) && r == e

Reply:
This test reflects my use-case pretty well, where the unshare -U and new{u,g}idmap is called by podman offcourse:
```
runuser foo -g bar -c "unshare -U sleep 10 & pid=\$!; \
        newgidmap \$pid 0 1001 1 1 1000000 1000; ret=\$?; \
	cat /proc/\$pid/gid_map >/tmp/test-gidmap/gid_map;
        kill \$pid; exit \$ret"
```

Reply:
Thanks.  I guess the only case I can imagine is:

user A used group Asharesdata to share data with user B.  Makes some
files owned by group Asharesdata which have group write perms, and
creates a setgid binary which allows user B to update it.

If the binary did a unshare, or could be made to do so, then userB
coudl run the setgid binary and quickly does a newuidmap+newgidmap
on it.  At this point B could map B's uid on the host to A's uid in
the namespace, potentially confusing it.

This seems a laughable scenario.  Let me think about it for one
more day and then I think I'll just take it as is.